contracts/NodeDelegator.sol

```diff
@@ -342,6 +392,167 @@ contract NodeDelegator is INodeDelegator, LRTConfigRoleChecker, PausableUpgradea
+        _requestWithdrawal(strategyAddress, strategyShares);
+
+        // account for the pending withdrawal as the shares are no longer accounted for in the EigenLayer strategy
+        pendingInternalShareWithdrawals[strategyAddress] += strategyShares;
+    }
+
+    /// @dev request the withdrawal of the LSTs from EigenLayer's underlying strategy.
+    /// @param strategy the address of the EigenLayer LST strategy to withdraw from
+    /// @param strategyShares the amount of EigenLayer strategy shares to redeem
+    /// @return withdrawalRoot the hash of the withdrawal data
+    function _requestWithdrawal(address strategy, uint256 strategyShares) internal returns (bytes32 withdrawalRoot) {
+        IStrategy[] memory strategies = new IStrategy[](1);
+        strategies[0] = IStrategy(strategy);
+
+        // Calculate how many EigenLayer strategy shares to redeem to get the requested asset amount
+        uint256[] memory shares = new uint256[](1);
+        shares[0] = strategyShares;
+
+        // request the withdrawal of the LSTs from EigenLayer
+        IDelegationManager.QueuedWithdrawalParams[] memory requests = new IDelegationManager.QueuedWithdrawalParams[](1);
+        requests[0] = IDelegationManager.QueuedWithdrawalParams(strategies, shares, address(this));
+        address delegationManagerAddress = lrtConfig.getContract(LRTConstants.EIGEN_DELEGATION_MANAGER);
+
+        // request the withdrawal from EigenLayer.
+        // Will emit the Withdrawal event from EigenLayer's DelegationManager which is needed for the claim.
+        withdrawalRoot = IDelegationManager(delegationManagerAddress).queueWithdrawals(requests)[0];
+    }
+
+    /// @notice Claims the previously requested withdrawal from EigenLayer's underlying strategy.
+    /// Transfers the withdrawn assets to the staker that requested the withdrawal.
+    /// Is only callable by the `LRTDepositPool` contract.
+    /// @param withdrawal the `withdrawal` data emitted in the `WithdrawalQueued` event from EigenLayer's
+    /// `DelegationManager` contract when `requestWithdrawal` was called on the `LRTDepositPool` contract by the staker.
+    /// @param staker the address of the staker requesting the withdrawal
+    /// @return asset address of the liquid staking tokens (LST) that were claimed.
+    /// @return assets the amount of LSTs received from the withdrawal
+    function claimWithdrawal(
+        IDelegationManager.Withdrawal calldata withdrawal,
+        address staker
+    )
+        external
+        onlyDepositPool
+        whenNotPaused
+        returns (address asset, uint256 assets)
+    {
+        // Make sure the staker requested this withdrawal
+        // and the withdrawal was not manipulated
+        bytes32 withdrawalRoot = _calculateWithdrawalRoot(withdrawal);
+        if (withdrawalRequests[withdrawalRoot] != staker) {
+            revert StakersWithdrawalNotFound();
+        }
```

Comment:
I'd sleep better at night if we deleted the `withdrawalRequests` after checking it, so that it's invalid on our end for further use. Of course in theory eigenlayer would block it, but if we are tracking it here too, I'd like to do it cleanly.

Reply:
I've gone with set `withdrawalRequests[withdrawalRoot]` to `0xdead` after a withdraw.

Reply:
Looks good!